### PR TITLE
[ML] Allow setting num_top_classes to a special value -1

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR allows setting `num_top_classes` parameter to a special value `-1`.
When `-1` is set, all the top classes will be reported in the analysis results, regardless the number.

Relates https://github.com/elastic/elasticsearch/issues/63306

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/default-distro

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public static Classification randomClassification() {
             .setEta(randomBoolean() ? null : randomDoubleBetween(0.001, 1.0, true))
             .setMaxTrees(randomBoolean() ? null : randomIntBetween(1, 2000))
             .setFeatureBagFraction(randomBoolean() ? null : randomDoubleBetween(0.0, 1.0, false))
-            .setNumTopFeatureImportanceValues(randomBoolean() ? null : randomIntBetween(0, Integer.MAX_VALUE))
+            .setNumTopFeatureImportanceValues(randomBoolean() ? null : randomInt())
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java:45]: Why was this changed?

Comment (przemekwitek) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java:45]: I wanted the client code to be able to pass `-1` value through.

**Comment by przemekwitek** [comment]
run elasticsearch-ci/1

**Comment by przemekwitek** [comment]
run elasticsearch-ci/packaging-sample-unix

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java:45]: But this is for `numTopFeatureImportanceValues`.

Comment (przemekwitek) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java]: Ooops, shame on me :(
Fixed.

**Review by szabosteve — APPROVED** [review]
Docs changes look good to me.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM
